Mathlib/MeasureTheory/Integral/CircleAverage.lean

```diff
@@ -0,0 +1,150 @@
+`circleAverage` defined here is a purely measure-theoretic average. It should not be confused with
+`CircleIntegral`, which is the path integral over the circle path. The relevant integrability
+property `circleAverage` is `CircleIntegrable`, as defined in
+`Mathlib.MeasureTheory.Integral.CircleIntegral`.
+
+Implementation Note: Like `circleMap`, `circleAverage`s are defined for negative radii. The theorem
+`circleAverage_congr_negRadius` shows that the average is independent of the radius' sign.
+-/
+
+open Filter Metric Real
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜] [NormedSpace 𝕜 E] [SMulCommClass ℝ 𝕜 E]
+  {f f₁ f₂ : ℂ → E} {c : ℂ} {R : ℝ} {a : 𝕜}
+
+namespace Real
+
+/-!
+# Definition
+-/
+
+variable (f c R) in
+/--
+Define `circleAverage f c R` as the average value of `f` on the circle with center `c` and radius
+`R`. This is a real notion, which should not be confused with the complex path integral notion
+defined in `circleIntegral` (integrating with respect to `dz`).
+-/
+noncomputable def circleAverage : E :=
+  (2 * π)⁻¹ • ∫ θ in (0)..2 * π, f (circleMap c R θ)
+
+lemma circleAverage_def :
+    circleAverage f c R = (2 * π)⁻¹ • ∫ θ in (0)..2 * π, f (circleMap c R θ) := rfl
+
+/-- Expression of `circleAverage´ in terms of interval averages. -/
+lemma circleAverage_eq_intervalAverage :
+    circleAverage f c R = ⨍ θ in (0)..2 * π, f (circleMap c R θ) := by
+  simp [circleAverage, interval_average_eq]
+
+/-- Interval averages for zero radii equal values at the center point. -/
+@[simp] lemma circleAverage_zeroRadius [CompleteSpace E] :
+    circleAverage f c 0 = f c := by
+  rw [circleAverage]
+  simp only [circleMap_zero_radius, Function.const_apply,
+    intervalIntegral.integral_const, sub_zero,
+    ← smul_assoc, smul_eq_mul, inv_mul_cancel₀ (mul_ne_zero two_ne_zero pi_ne_zero),
+    one_smul]
+
+/--
+Expression of `circleAverage´ with arbitrary center in terms of `circleAverage` with center zero.
+-/
+lemma circleAverage_fun_add :
+    circleAverage (fun z ↦ f (z + c)) 0 R = circleAverage f c R := by
+  unfold circleAverage circleMap
+  congr
+  ext θ
+  simp only [zero_add]
+  congr 1
+  ring
+
+/-!
+## Congruence Lemmata
+-/
+
+/-- Circle averages do not change when shifting the angle. -/
+lemma circleAverage_eq_integral_add (η : ℝ) : circleAverage f c R =
+    (2 * π)⁻¹ • ∫ (θ : ℝ) in (0)..2 * π, f (circleMap c R (θ + η)) := by
+  rw [intervalIntegral.integral_comp_add_right (fun θ ↦ f (circleMap c R θ))]
+  have t₀ : (fun θ ↦ f (circleMap c R θ)).Periodic (2 * π) :=
+    fun x ↦ by simp [periodic_circleMap c R x]
+  have := t₀.intervalIntegral_add_eq 0 η
+  rw [zero_add, add_comm] at this
+  rw [zero_add]
+  simp only [circleAverage, mul_inv_rev]
+  congr
+
+/-- Circle averages do not change when replacing the radius by its negative. -/
+theorem circleAverage_congr_neg_radius :
+    circleAverage f c R = circleAverage f c (-R) := by
```

Comment:
```suggestion
@[simp] theorem circleAverage_neg_radius :
    circleAverage f c (-R) = circleAverage f c R := by
```
More complicated expression to the left. And like that this becomes a good simp lemma.